fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineScrubber.java

```diff
@@ -0,0 +1,1046 @@
+/*
+ * OnlineScrubber.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.async.AsyncUtil;
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.RecordMetaData;
+import com.apple.foundationdb.record.RecordMetaDataProvider;
+import com.apple.foundationdb.record.RecordStoreState;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.MetaDataException;
+import com.apple.foundationdb.record.metadata.RecordType;
+import com.apple.foundationdb.record.provider.common.RecordSerializer;
+import com.apple.foundationdb.record.query.plan.synthetic.SyntheticRecordPlanner;
+import com.apple.foundationdb.subspace.Subspace;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+@API(API.Status.UNSTABLE)
+public class OnlineScrubber implements AutoCloseable {
+    @Nonnull
+    private static final Logger LOGGER = LoggerFactory.getLogger(OnlineScrubber.class);
+
+    @Nonnull private final IndexingCommon common;
+    @Nonnull private final FDBDatabaseRunner runner;
+    @Nonnull private final Index index;
+
+    @Nullable
+    private IndexingBase indexer = null;
```

Comment:
Is this used?

Reply:
Mmm... it's not there anymore. 
Were you looking at the pre-cleanup version? Should I have squashed it?

Reply:
Ah, yes, sorry, I thought I'd confirmed that it was in the final clean up version, too, but my mistake.